### ISSUE TITLE
About Tinybox & Multi-GPU Training & Bandwidth Speed!

### ISSUE THREAD

**Issue by faruknane** [tinygrad/tinygrad#9094]
Hi,

I couldn't find documents for strategies around multi GPU training solutions based on tinygrad. It has come to my attention that the tinybox only supports pcie 4 bandwidth. So far, I also trained some models using 2 x 5090 with the pytorch 2.7 preview version (pytorch 2.6 doesn't support 5090). The transfer bandwidth is not good even for PCI 5. However, I don't see a nice improvement over 1 x 5090 unless you accumulate gradients and try to transfer them less and less between gpus or compress them, etc. However, when that's the case, your speed increases but since your effective batch size increases as well, you make fewer grad updates and hence the training speed doesn't improve (since you accumulate grads if you don't accumulate grads, otherwise your speed wouldn't be good). One way to deal with that would be increasing the learning rate, however, you can't linearly scale lr that is not practical in real life. 

Is there a plan to support tinybox in terms of multi GPU training strategy? How would you consider optimizing the training pipeline for slow bandwidth?

Nvidia strategically removed the NVLink from 4090's. No consumer GPU supports NVLink anymore. Do you have any idea to deal with this?

Thanks,

**Comment by geohot** [comment]
Not tinygrad issue
